src/main/java/org/ms/zk/plugin/matcher/role/RoleMatcher.java

```diff
@@ -0,0 +1,33 @@
+package org.ms.zk.plugin.matcher.role;
+
+import com.unboundid.ldap.sdk.Attribute;
+import com.unboundid.ldap.sdk.LDAPException;
+import org.ms.zk.plugin.utils.LdapQuery;
+import org.ms.zk.plugin.matcher.Matcher;
+
+import java.io.IOException;
+import java.util.Collection;
+
+import static java.lang.System.getenv;
+import static org.ms.zk.plugin.utils.Configuration.get;
+
+public abstract class RoleMatcher extends Matcher {
+  public boolean matches(String id, String aclExpr) throws IOException, LDAPException {
+
+    String baseDN = get(baseDN(), getenv("TREADMILL_LDAP_SUFFIX"));
+    String filter = get(filter(), getenv("TREADMILL_CELL"));
+
+    Collection<Attribute> attributes = LdapQuery.getAttributes(baseDN, filter);
+
+    return attributes.stream().anyMatch(
+      a ->
+        a.getName().startsWith(pattern())
+          && a.getValue().equals(id.split("@")[0]));
```

Comment:
is this because we do not store realm of the host in the db? i think you should match on realm as well.